psa-crypto/src/types/key.rs

```diff
@@ -113,6 +113,36 @@ impl Attributes {
         }
     }
 
+    /// Check if a key has permission to sign a message
+    pub fn is_message_signable(self) -> bool {
+        self.policy.usage_flags.sign_hash & self.policy.usage_flags.sign_message
```

Comment:
Shouldn't it be `|` here instead of `&`?

From the [`sign_hash`](https://parallaxsecond.github.io/parsec-book/parsec_client/operations/psa_key_attributes.html#sign-hash-usage) explanation:

> This flag automatically sets sign_message: if an application sets the flag sign_hash when creating a key, then the key always has the permissions conveyed by sign_message. For a key pair, this concerns the private key.

It means that a key can either have (ideally, this is currently not implemented, will be in #82):
1. `sign_hash` and `sign_message` to `false`
2. `sign_hash` and `sign_message` to `true`
3. `sign_hash` to `false` and `sign_message` to `true`

`&` would mean that 3 is not possible?

And same for `verify`. Not 100% sure so feel free to correct me!

Reply:
Yeah, that's true, I mis-remembered it the other way around - if you have `sign_message` you should also have `sign_hash` (since you're hashing the message and signing that...).

Reply:
So it is possible scenario for 3. to exist? If yes then it make sense to change it to `|`.
As Ionut metioned and from point of view of cryptoauthlib you always need to be able to sign hash to sign message.

Reply:
> So it is possible for scenario 3. to exist?

From the PSA Crypto API, yes it is! The scenario `sign_hash` to `true` and `sign_message` to `false` is not possible.